calculating_n50_assemblies.py

```diff
@@ -3,60 +3,77 @@
+    output_file = open(os.path.join(out_dir,'summary_statistics.txt'),'w')
     fasta_ext = ['fasta','fna','ffn','faa','frn']
-
-    switch = 'off'
+    n50_array = [] 
+    n50_array_log = []
     for f_name in os.listdir(in_dir):
-        if f_name.endswith('.gz'):
+        if fnmatch.fnmatch(f_name, '*.*'):
             array_f = f_name.split('.')
-            if array_f[-2] in fasta_ext:
-                file = os.path.join(in_dir,f_name)
+            file = os.path.join(in_dir,f_name)
+            if (array_f[-1] == 'gz' and array_f[-2] in fasta_ext):
                 with gzip.open(file, mode = 'rt') as open_file:
-                    n50_calc(open_file, output_file)
-                switch = 'on'
-        else:
-            array_f = f_name.split('.')
-            if array_f[-1] in fasta_ext:
-                file = os.path.join(in_dir,f_name)
+                    n50_calc(open_file, n50_array, n50_array_log)
+            elif array_f[-1] in fasta_ext:
                 with open(file, mode = 'r') as open_file:
-                    n50_calc(open_file, output_file)
-                switch = 'on'
-
-    if switch == 'off':
+                    n50_calc(open_file, n50_array, n50_array_log)
+                   
+    if not n50_array:
         sys.exit("No zipped or unzipped fasta assemblies are in the directory. Exiting..")
+                   
+    else:
+        n50_summary(n50_array, n50_array_log, output_file, out_dir)
+
+def n50_calc(open_file = None, n50_array = None, n50_array_log = None):
+    contig_length_dict = {}
+    for line in open_file:
+        x = re.findall(r'>(.+?)\s(.+?)\s(.+?)\s', line)
+        if len(x) > 0:
+            contig_name = x[0][0]
+            org_name = x[0][1]+" "+x[0][2]
+            contig_length = 0
+        else:
+            contig_length += len(line.rstrip('\n'))
+            contig_length_dict[contig_name] = contig_length
 
-def n50_calc(open_file = None, output_file = None):
-        contig_length_dict = {}
-        for line in open_file:
-            assembly = re.findall(r'^>([A-Z]+?0\d).+',line)
-            if len(assembly) > 0:
-                assembly_name = assembly[0] 
-            x = re.findall(r'^>(.+?)\s', line)
-            if len(x) > 0:
-                contig_name = x[0]
-                contig_length = 0
-            else:
-                contig_length += len(line.rstrip('\n'))
-                contig_length_dict[contig_name] = contig_length
+    total_assembly_len = 0
+    for key in contig_length_dict:
+        total_assembly_len += contig_length_dict[key]
+    l50_len = total_assembly_len/2
+    rev_sorted_contig_length_dict = sorted(contig_length_dict.items(), key=operator.itemgetter(1),reverse=True)
+    temp_sum = 0
+    for item in rev_sorted_contig_length_dict:
+        temp_sum += item[1]
+        if temp_sum >= l50_len: 
+            n50_array.append(item[1])
+            n50_array_log.append(math.log(item[1],10))
+            break
+    n50_calc.org_name = org_name
```

Comment:
Do you understand what you are doing here with `n50_calc.org_name = org_name`?

Reply:
defining a function attribute to use later outside the function?

Reply:
Ok that is correct, I would advise against doing that though as it is not normal practice. Instead I would add a return variable or just get the user to supply the organism via a command line option

Reply:
I did via command line